pkg/registry/networking/clustercidrconfig/strategy.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustercidrconfig
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/networking"
+	"k8s.io/kubernetes/pkg/apis/networking/validation"
+)
+
+// clusterCIDRConfigStrategy implements verification logic for ClusterCIDRConfigs.
+type clusterCIDRConfigStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating clusterCIDRConfig objects.
+var Strategy = clusterCIDRConfigStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// NamespaceScoped returns false because all clusterCIDRConfigs do not need to be within a namespace.
+func (clusterCIDRConfigStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate clears the status of a ClusterCIDRConfig before creation.
+func (clusterCIDRConfigStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	clusterCIDRConfig := obj.(*networking.ClusterCIDRConfig)
+	clusterCIDRConfig.Generation = 1
```

Comment:
Do we need to set this, if it is immutable?

Reply:
Removed, yes currently CCC are immutable so not required to be set